### DISCUSSION TITLE
Quarto and Zotero (with Better BibTeX) -- will it be improved?

### DISCUSSION THREAD

**Discussion by jhchou** [quarto-dev/quarto-cli#2660] [Q&A]
_(Apologies for the non-specific nature of this post -- it's not a tight, single issue -- but I had posted it on the RStudio community forums in this conversational format, and was asked to cross-post it here.)_

I'm trying to do some academic writing with Quarto documents which requires proper citations. It's great to have (some) support for Zotero (and I use Better BibTeX as well), but the implementation thus far is rather rough. I know that's the price for being a bit on the "bleeding edge" of development, with Quarto, but are there plans to improve support?

Some issues I've run into include:

* only being able to Insert a citation easily from the Visual, and not Source, editor?
* search functionality is truly truly terrible
* MULTIPLE copies of citations can end up being added to the .bib file
* very non-intuitive, and I think probably brittle, use of Better BibTeX (hard to find, doesn't seem to always work. In case anyone else couldn't figure it out: Preferences -> R Markdown -> Citations -> "Use Better BibTeX..."

If Quarto is being pushed as a pretty robust technical writing / publishing platform, I wonder whether it would be beneficial to polish the citation support?

**Comment by dragonstyle** [comment]
Thanks for taking the time to write this up - we definitely want to continue improving the citations features. Some of the items we're aware (e.g. it not working in the source editor), but others would be helpful to learn more about:

- Can you say more about your experience with search and the ways that it isn't working well for you?
- We definitely are doing work to try to prevent duplication in bibliographies, but it sounds like in your case this isn't working well at all. Do you happen to have an example I could use to investigate / debug (e.g. the steps, or perhaps the citations that are leading to duplication)?
- We're aware of Better BibTeX being pretty had to find in options, but can you say a bit more about what doesn't work about it?

**Comment by rgaiacs** [comment]
Hi @jhchou, Thanks for the message. Can you share the link to your post in the R Studio community forum? I'm also trying to do some academic writing with Quarto. :-D

> search functionality is truly truly terrible

I believe you are talking about the search using

![Screenshot from 2022-09-29 09-34-44](https://user-images.githubusercontent.com/1506457/192918578-bd11ea35-5476-48db-8965-5c53ca045c76.png)

In my workflow, I use Zotero to organise the references and Better BibTeX to continuously update my `references.bib`.

**Reply by KleinWurm** [reply]
Hi,
Here is the link to the post in the R Comminity Forum: https://community.rstudio.com/t/quarto-and-zotero-with-better-bibtex-will-it-be-improved/148462

I am starting an academic work with the idea of using Zotero and Quarto. I came across the problem of updating the .bib file (and eventually the `@refs` in the .qmd file) when changes are made to the item in Zotero.
Example:
1. I have a document called "A very good paper" in my Zotero Library.
2. I cite this paper in the .qmd document: `blablabla @authdate`
3. I change something in the metadata of "A very good paper", is it possible to reflect this changes on the .bib document used by .qmd to edit the bibliography accordingly?

In your comment, you mention that "in [your] workflow, [you] use Zotero to organise the references and Better BibTeX to continuously update [your] references.bib" could you please elaborate on that or point some tutorials?

**Comment by mccarthy-m-g** [comment]
Since you’re using R, I’d recommend using the {rbbt} package for citations with Zotero + Quarto: <https://github.com/paleolimbot/rbbt>. It addresses all the issues you’re having with RStudio’s built-in Zotero support.

**Comment by stefanocoretta** [comment]
Two more things (I'm thinking more VS Code/Positron):

- As others have pointed out in other threads, some fields are not correctly imported when using BibTeX as the format (I haven't tried with other formats). For example, "preprints" don't get a year when imported so the citation key is just the surname.

- Citation keys keep non-ASCII characters. So Munafó 2020 gets a key munafó2020 while in Zotero BetterBibtex is set to strip non-ASCII characters.
